ecs-agent/netlib/model/ecscni/nsutil_linux.go

```diff
@@ -0,0 +1,154 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//	http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+//go:build !windows
+// +build !windows
+
+package ecscni
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"runtime"
+	"strings"
+	"syscall"
+
+	cnins "github.com/containernetworking/plugins/pkg/ns"
+	"github.com/pkg/errors"
+)
+
+const (
+	// dnsNameServerFormat defines the entry of nameserver in resov.conf file
+	dnsNameServerFormat = "nameserver %s\n"
+	// dnsNameServerFormat defines the entry of search in resov.conf file
+	dnsSearchDomainFormat = "search %s\n"
+)
+
+// NetNSUtils provides some basic methods for agent to deal with network namespace
+type NetNSUtil interface {
+	// NewNetNS creates a new network namespace in the system
+	NewNetNS(nsPath string) error
+	// DelNetNS deletes the network namespace from the system
+	DelNetNS(nsPath string) error
+	// GetNetNSPath cretes the network namespace path from named namespace
+	GetNetNSPath(nsName string) string
+	// GetNetNSName extract the ns name from the netns path
+	GetNetNSName(nsPath string) string
+	// NSExists checks if the given ns path exists or not
+	NSExists(nsPath string) (bool, error)
+	// ExecInNSPath invokes the function in the given network namespace
+	ExecInNSPath(nsPath string, cb func(cnins.NetNS) error) error
+	// BuildResolvConfig constructs the content of dns configuration file resolv.conf
+	BuildResolvConfig(nameservers, searchDomains []string) string
+}
+
+type netnsutil struct {
+}
+
+func NewNetNSUtil() NetNSUtil {
+	return &netnsutil{}
+}
+
+// NewNetNS create a new network namespace with given path
+func (*netnsutil) NewNetNS(nspath string) error {
```

Comment:
I understand that tests for these functions are missing in the original repo. As an improvement, can we add some tests for these here?

Reply:
Sure I will take a look

Reply:
Added!